### PR TITLE
Add unit tests for thread summary feature

### PR DESCRIPTION
**Pull request by hamza221**
ref https://github.com/nextcloud/mail/issues/8508

### CODE REVIEW

File: tests/Unit/Controller/ThreadControllerTest.php [hunk 1]
```diff
@@ -237,4 +237,48 @@ public function testDeleteTrash(): void {
 
 		$this->assertEquals(Http::STATUS_OK, $response->getStatus());
 	}
+
+	public function testSummarizeThread(): void {
+		$mailAccount = new MailAccount();
+		$mailAccount->setId(1);
+		$this->accountService
+			->method('find')
```
Comment (ChristophWurst) [inline tests/Unit/Controller/ThreadControllerTest.php:245]: nit: if you add `->with(...)` you can assert it's looking for the expected id

File: tests/Unit/Service/AiIntegrationsServiceTest.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		parent::setUp();
+		$this->container = $this->createMock(ContainerInterface::class);
+		$this->manager = $this->createMock(IManager::class);
+
+		$this->aiIntegrationsService = new AiIntegrationsService(
+			$this->container
+		);
+	}
+
+	public function testSummarizeThreadNoBackend() {
```
Comment (ChristophWurst) [inline tests/Unit/Service/AiIntegrationsServiceTest.php:59]: add a test where `get` throws

File: tests/Unit/Controller/ThreadControllerTest.php (hunk 1 above)
Comment (hamza221) [reply tests/Unit/Controller/ThreadControllerTest.php:245]: you mean ->with() on the accountService?

Comment (ChristophWurst) [reply tests/Unit/Controller/ThreadControllerTest.php:245]: yes

Comment (ChristophWurst) [reply tests/Unit/Controller/ThreadControllerTest.php:245]: ``$this->accountService->expect(self::once())->method('find')->with(1)->willReturn(new Account($mailAccount));``

then you know
1) it's been called exactly once
2) it's been called with the expected inputs

**Comment by ChristophWurst** [comment]
26 fails with `Object of class OCA\Mail\Exception\ServiceException could not be converted to int`

**Review by kesselb — APPROVED** [review]

**Pull request merged**